pkgs/applications/networking/cluster/argocd/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, buildGoModule, fetchFromGitHub, packr }:
+
+buildGoModule rec {
+  pname = "argocd";
+  version = "1.4.2";
+
+  goPackagePath = "github.com/argoproj/argo-cd";
+
+  src = fetchFromGitHub {
+    owner = "argoproj";
+    repo = "argo-cd";
+    rev = "v${version}";
+    sha256 = "01vsyrks1k5yfvrarv8ia0isr7snilr21b7lfiy860si82r2r8hj";
+  };
+
+  modSha256 = "1qivg7yy7ymmgkrvl365x29d8jnsphbz18j1ykgwwysyw3n4jkdg";
+
+  nativeBuildInputs = [ packr ];
+
+  patches = [ ./use-go-module.patch ];
+ 
+  preBuild = ''
+    # run packr to embed assets
```

Comment:
It's better to have comments in Nix rather than shell scripts. Changing a comment embedded in a shell script will require a rebuild of the package.